recipe-server/client/control_new/state/app/session/actions.js

```diff
@@ -0,0 +1,71 @@
+/* eslint import/prefer-default-export: "off" */
+
+import { List, Map } from 'immutable';
+
+import {
+  SESSION_INFO_RECEIVE,
+  SESSION_INFO_HISTORY_VIEW,
+  REQUEST_SEND,
+  REQUEST_SUCCESS,
+} from 'control_new/state/action-types';
+
+const STORAGE_KEY = 'normandy-sesh-info';
+
+export function fetchSessionInfo() {
+  return async dispatch => {
+    const requestId = 'fetch-session-info';
+    dispatch({
+      type: REQUEST_SEND,
+      requestId,
+    });
+
+    // Read the stringified object and convert it (and its nested objects) into
+    // proper immutable objects.
+    let savedHistory = JSON.parse(localStorage.getItem(STORAGE_KEY) || '{}');
```

Comment:
Should this use `localForage` instead of `localStorage`? We already have it as a dependency, and you already have all the async machinery in these action creators.